### PR TITLE
upgrade to pytorch 0.4.1 + make work with python 3.7 (but still 3.6 also)

### PR DESCRIPTION
**Pull request by joelgrus**
tested under (3.6 + 0.4), (3.6 + 0.4.1), (3.7 + 0.4.1)

(I am pretty sure that 3.7 requires 0.4.1)

nasty subtlety where Python 3.7 changed an undocumented feature I was relying on to identify annotations as Dicts or Lists.

also, I added a script to check whether setup.py and requirements.txt are in sync, it's kind of hacky, but it found me a discrepancy already

### CODE REVIEW

**Comment by matt-gardner** [comment]
Seems fine to just lower the precision of that test.  6 decimals is plenty.

File: setup.py [hunk 1]
```diff
@@ -101,7 +101,7 @@
       packages=find_packages(exclude=["*.tests", "*.tests.*",
                                       "tests.*", "tests"]),
       install_requires=[
-          'torch==0.4.0',
+          'torch==0.4.1',
```
Comment (DeNeutoy) [inline setup.py]: >=0.4.0?

Comment (joelgrus) [reply setup.py]: what's the rationale for allowing 0.4.0 still?

Comment (matt-gardner) [reply setup.py]: That's also a valid point, if the version matters for our support. I could go either way here.

Comment (DeNeutoy) [reply setup.py]: For minor versions of the key library that we depend on, this seems a bit inflexible. But, if you feel strongly, go ahead.

Comment (matt-gardner) [reply setup.py]: The only other thing I'd add is that if we have a lower bound, we should have an upper bound too.  Not sure if that should be `0.4.2`, `0.5.0`, or `1.0.0`.

**Comment by huntzhan** [comment]
Hi Team, any plan for a new release of this bugfix?

**Pull request merged**